assets/ts/api/request.ts

```diff
@@ -17,18 +17,30 @@ export function request<dataType>({ endpoint, query, body, method, headers }: re
         Object.entries(query).forEach(([key, value]) => url.searchParams.append(key, value));
     }
 
-    return fetch(url, {
+    // TODO: catch 429 and make retries
+
+    const response = await fetch(url, {
         method: method ?? "GET",
         headers: {
             Authorization: `Bearer ${auth}`,
             "Content-Type": "application/json",
             ...headers,
         },
         body: body ? JSON.stringify(body) : undefined,
-    }).then((response) => {
-        return response.json();
-    }).then<dataType>((data) => {
-        if (data.error) {
+    });
+
+    return response;
+}
+
+export async function request<DataType>(parameters: requestParameter, noContent: any = null) {
```

Comment:
naming vorschlag: noContent -> fallback.
Aber das ist hier an komischer stelle. Wollen wir den return type nicht einfach `Promise<DataType | null>` machen?
Dann ist der fallback transparenter.

Reply:
fallback wäre zu missführend. noContent beschreibt genau, dass es bei 204 No-Content komment. fallback würde bedeuten, dass es kommt bei !response.ok

Reply:
Ist es nicht etwas verwirrend. component-spezisfische Logik in einer generischen komponente zu haben? Zudem muss noContent ja ein Promise<> sein wegen dem promise chaining. Da sehe ich viel fehlerpotenzial.
Was hältst du von `if (status = 204) return new Promise.rejected(new Error("204"))`? dann kann bei dem aufrufen von request() ein .catch() block erstellt werden um den case abzufangen.
Man müsste natürlich noch gucken, dass promise chaining nicht in den weg kommt...

Reply:
1. da das ein http-status-code ist ist es eig. keine component spezifische logik
2. noContent muss kein Promise sein für das Promise chaining.
3. mit Promise.reject muss ich noch überlegen, aber auf keine fall mit .catch(). .catch() ist für fehler und das ist eindeutig keine. in diesem fall müsste man dann .then(onSuccess, onRejected) benutzen

Reply:
Ja, der 204er ist nicht unbedingt Komponenten spezifisch. Er ist aber sehr untypisch und ich finde es würdigt nicht einen extra Parameter für die Funktion (Da es pro Parameter viel komplexer ist zu verstehen). Wir können ja nicht für jeden http-code ein return value mitgeben.

Reply:
Aber noContent.then() wird ja aufgerufen... löst das nicht wenigstens ein console error aus?

Reply:
Nur weil es ein 200 http Code ist, heißt nicht, dass wir es nicht Symantisch als logischen Fehler behandeln können. So wie ich das verstehe fragen wir:
"Was ist der Player Status?" 
Und Spotify entgegnet:
"Alles gut, es gibt keinen Player Status, er läuft nämlich nicht."

Das ist für uns ja ein Problem, da wir den Player dann nicht anzeigen können. Und wenn man mit Problemen umgehen muss, finde ich ein .catch() Block eigentlich sehr passend, oder?

Reply:
> Aber noContent.then() wird ja aufgerufen... löst das nicht wenigstens ein console error aus?

Da hast du das Promise-System von js missverstanden. wenn du einen wert übergibst, dann wrappt Promise.then() diesen wert automatisch.
Dass heißt, dass es nicht `noContent.then()` ist, sondern `Promise<noContent>.then()`

Reply:
Ahhh, stimmt. Sorry, ich bin noch nicht so erfahren mit promises. Danke für die Erklärung